### PR TITLE
[refactor] API Provider request steps

### PR DESCRIPTION
**Pull request by mmuzikar**
//test: `@api-provider`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Create and delete 
Connection - CRUD | Kebab menu 
Connection - CRUD | Kebab menu edit and view 
API Provider Integration | Create API provider from url spec todo-app 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.json 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.swagger.yaml 
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Publishing an integration without any operation should not be possible 
API Provider Integration | Clicking Go To Operation List does not discard progress in API Provider 
API Provider Integration | Clicking back button from Apicurio should work consistently in API Provider 
API Provider Integration | API Provider operation with simple return type 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | Create API provider from url spec todo-app 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.json 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.swagger.yaml 
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Publishing an integration without any operation should not be possible 
API Provider Integration | Clicking Go To Operation List does not discard progress in API Provider 
API Provider Integration | Clicking back button from Apicurio should work consistently in API Provider 
API Provider Integration | API Provider operation with simple return type 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | Publishing an integration without any operation should not be possible 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | Clicking back button from Apicurio should work consistently in API Provider 
API Provider Integration | API Provider operation with simple return type 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Create and delete 
Connection - CRUD | Kebab menu 
Connection - CRUD | Kebab menu edit and view 
API Provider Integration | Create API provider from url spec todo-app 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.json 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.swagger.yaml 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type 
Integration info | Check error for integration with the same name 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
`API Provider Edit OpenAPI - add operation` will get addressed when we fully start to support OpenAPI
`API Provider Integration | API Provider export roundtrip` needs investigation, but it passes locally

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type  - known issue
Integration - DB to DB | Read and update operations 
Integration info | Check error for integration with the same name 
Integration info | Check error for integration which has same name after update 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: ui-tests/src/test/java/io/syndesis/qe/fragments/common/form/ApiSpecificationForm.java [hunk 1]
```diff
@@ -48,7 +54,21 @@ public void uploadFileFromUrl(String url) {
         $(Input.URL).shouldBe(visible).sendKeys(url);
     }
 
-    public void createFromScratch() {
-        $(Input.CREATE_FROM_SCRATCH).shouldBe(visible).click();
+    public void createFromScratch(String url) {
```
Comment (asmigala) [inline ui-tests/src/test/java/io/syndesis/qe/fragments/common/form/ApiSpecificationForm.java]: The argument should probably not be called `url`.

File: ui-tests/src/test/resources/features/integrations/api-provider.feature [hunk 2]
```diff
@@ -76,7 +76,7 @@ Feature: API Provider Integration
     Then check visibility of page "Upload API Provider Specification"
 
     # hacky way to reuse the existing step
-    When create API Provider spec from scratch .
+    When create API Provider spec from scratch
```
Comment (asmigala) [inline ui-tests/src/test/resources/features/integrations/api-provider.feature]: Will this work with the newly refactored `createFromScratch` method? Looks like should specify the openapi version here.

Comment (mmuzikar) [reply ui-tests/src/test/resources/features/integrations/api-provider.feature]: If url is null, the default => 2.0 version will be used, but I won't be lazy anymore and change it so it's obvious

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
@asmigala What do you feel like with the default 2.0 version? We can entirely drop it and make it mandatory to always supply the spec version.

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | Create from scratch 
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | Publishing an integration without any operation should not be possible 
API Provider Integration | Clicking back button from Apicurio should work consistently in API Provider 
API Provider Integration | API Provider operation with simple return type  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider DELETE  - known issue
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider operation with simple return type  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Pull request merged**